satpy/tests/reader_tests/test_satpy_cf_nc.py

```diff
@@ -85,25 +88,24 @@ def setUp(self):
 
     def test_write_and_read(self):
         """Save a file with cf_writer and read the data again."""
-        # '{testin}-{sensor}-{start_time:%Y%m%d%H%M%S}-{end_time:%Y%m%d%H%M%S}.nc'
         filename = 'testingcfwriter{:s}-viirs-mband-20201007075915-20201007080744.nc'.format(
             datetime.utcnow().strftime('%Y%j%H%M%S'))
-        self.scene.save_datasets(writer='cf',
-                                 filename=filename,
-                                 header_attrs={'instrument': 'avhrr'},
-                                 engine='h5netcdf',
-                                 flatten_attrs=True,
-                                 pretty=True)
-        scn_ = Scene(reader='satpy_cf_nc',
-                     filenames=[filename])
-        scn_.load(['image0', 'image1', 'lat'])
-        self.assertTrue(np.all(scn_['image0'].data == self.scene['image0'].data))
-        self.assertTrue(np.all(scn_['lat'].data == self.scene['lat'].data))  # lat loaded as dataset
-        self.assertTrue(np.all(scn_['image0'].coords['lon'] == self.scene['lon'].data))  # lon loded as coord
         try:
-            os.remove(filename)
-        except PermissionError:
-            pass
+            self.scene.save_datasets(writer='cf',
+                                     filename=filename,
+                                     header_attrs={'instrument': 'avhrr'},
+                                     engine='h5netcdf',
+                                     flatten_attrs=True,
+                                     pretty=True)
+            scn_ = Scene(reader='satpy_cf_nc',
+                         filenames=[filename])
+            scn_.load(['image0', 'image1', 'lat'])
+            self.assertTrue(np.all(scn_['image0'].data == self.scene['image0'].data))
+            self.assertTrue(np.all(scn_['lat'].data == self.scene['lat'].data))  # lat loaded as dataset
+            self.assertTrue(np.all(scn_['image0'].coords['lon'] == self.scene['lon'].data))  # lon loded as coord
+        finally:
```

Comment:
Hm, never seen a try without an except. So any exceptions raise before `finally` is kind of ignored with this approach? Instead of
```
except:
  pass
```

Reply:
No, the exception isn't ignored, but 'finally' ensures its part of the code is executed before raising the exception.